PythonVisualizations/OrderedArray.py

```diff
@@ -68,40 +68,75 @@ def createIndex(self, index,  # cell
     def insert(self, val):
         callEnviron = self.createCallEnvironment()
         self.startAnimations()
-        j = self.find(val)  # Find where item should go
-
+       # j = self.find(val)  # Find where item should go
+        k=len(self.list)
         self.list.append(drawable(None))
         
         indexK = self.createIndex(len(self.list) -1, 'k', level=-1) # create "k" arrow
         callEnviron |= set(indexK)  
         
-        for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
-            
+       # for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
+        while 0 < k and self.list[k-1].val > val: # over items
+
             self.wait(1)
             for item in indexK:
                 self.canvas.move(item, -self.CELL_SIZE, 0)  # Move "k" arrow  
```

Comment:
Using the `wait(1)` followed by the `self.canvas.move()` for the indexK items means the animation stops for a second or more followed by a jump of the index by CELL_SIZE.  If you replace lines 81-83 with `self.moveItemsBy(indexK, (-self.CELL_SIZE, 0), sleepTime=0.1)`, it will move the same items in 10 steps over 1 sec.  That makes the animation smoother, and the users see something happening, rather than a pause that might look like it was stuck.  A less obvious effect of using that call is that there are 10 shorter wait periods, which is when the system can look for presses of Pause or Stop.  Having more frequent calls for short duration to `wait()` means the controls are more responsive.